choreolib/src/main/native/include/choreo/trajectory/Trajectory.h

```diff
@@ -0,0 +1,325 @@
+// Copyright (c) Choreo contributors
+
+#pragma once
+
+#include <algorithm>
+#include <optional>
+#include <ranges>
+#include <string>
+#include <vector>
+
+#include <wpi/json_fwd.h>
+
+#include "choreo/trajectory/DifferentialSample.h"
+#include "choreo/trajectory/EventMarker.h"
+#include "choreo/trajectory/SwerveSample.h"
+#include "choreo/trajectory/TrajectorySample.h"
+
+namespace choreo {
+template <TrajectorySample SampleType>
+/**
+ * A trajectory loaded from Choreo.
+ *
+ * @param <SampleType> DifferentialSample or SwerveSample.
+ */
+class Trajectory {
+ public:
+  /**
+   * Constructs a Trajectory with defaults
+   */
+  Trajectory() = default;
+  /**
+   * Constructs a Trajectory with the specified parameters.
+   *
+   * @param name The name of the trajectory.
+   * @param samples The samples of the trajectory.
+   * @param splits The indices of the splits in the trajectory.
+   * @param events The events in the trajectory.
+   */
+  Trajectory(const std::string& name, const std::vector<SampleType>& samples,
```

Comment:
Use `std::string_view` from `#include <string_view>` instead of `const std::string&`.